mlir/parser_transformer.py

```diff
@@ -221,4 +236,31 @@ def suffix_id(self, *suffix):
     affine_symbol = lambda self, value: value[0]
     semi_affine_symbol = lambda self, value: value[0]
 
+    ###############################################################
+    # MLIR file
+
+    def only_functions_and_definitions_file(self, defns_and_fns):
+        assert isinstance(defns_and_fns, list)
+        assert all(isinstance(el, tuple) for el in defns_and_fns)
+        defns = sum([defns for defns, fns in defns_and_fns], [])
+        fns = sum([fns for defns, fns in defns_and_fns], [])
```

Comment:
strange syntax to create a flat list. why?

Reply:
I could change this to `itertools.chain(*list_of_lists)`. Please let me know if you have something simpler in mind.

Reply:
It's probably fine :)